lib/cocoapods/generator/app_target_helper.rb

```diff
@@ -116,26 +113,23 @@ def self.add_swift_version(target, swift_version)
       # @param  [Symbol] platform
       #         the platform of the target. Can be `:ios` or `:osx`, etc.
       #
-      # @param  [Boolean] use_frameworks
-      #         whether to use frameworks or not when generating the contents of the import file.
-      #
       # @param  [String] name
       #         The name of the folder to use and save the generated main file.
       #
       # @return [Pathname] the new source file that was generated.
       #
-      def self.create_app_import_source_file(project, pod_target, platform, use_frameworks, name = 'App')
-        language = pod_target.uses_swift? ? :swift : :objc
+      def self.create_app_import_source_file(project, pod_target, platform, name = 'App')
+        language = [pod_target, *pod_target.recursive_dependent_targets].any?(&:uses_swift?) ? :swift : :objc
 
         if language == :swift
           source_file = project.path.dirname.+("#{name}/main.swift")
           source_file.parent.mkpath
-          import_statement = use_frameworks && pod_target.should_build? ? "import #{pod_target.product_module_name}\n" : ''
+          import_statement = pod_target.should_build? && pod_target.defines_module? ? "import #{pod_target.product_module_name}\n" : ''
```

Comment:
With Swift static libraries I believe this was outdated. We should attempt to import the module if the target supports it. No longer scoped to frameworks.

Is this correct?